unified-scheduler-pool/src/lib.rs

```diff
@@ -1487,7 +1548,9 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
                     session_result_sender
                         .send(result_with_timings)
                         .expect("always outlived receiver");
-                    state_machine.reinitialize();
+                    if matches!(scheduling_mode, BlockVerification) {
+                        state_machine.reinitialize();
```

Comment:
so, `state_machine` won't be reinitialized for BlockProduction anymore. along with this, proper oom mitigation will be addressed by later pr, while that concern is temporarily becoming more prominent.